examples/api/lib/widgets/scroll_view/listview_select.1.dart

```diff
@@ -0,0 +1,231 @@
+    return const MaterialApp(
+      title: _title,
+      home: ListTileSelectExample(),
+    );
+  }
+}
+
+class ListTileSelectExample extends StatefulWidget {
+  const ListTileSelectExample({Key? key}) : super(key: key);
+
+  @override
+  ListTileSelectExampleState createState() => ListTileSelectExampleState();
+}
+
+class ListTileSelectExampleState extends State<ListTileSelectExample> {
+  bool isEditMode = false;
+  final int listLength = 30;
+  late List<bool> _selected;
+  bool _selectAll = false;
+  bool _isGridMode = false;
+
+  @override
+  void initState() {
+    super.initState();
+    _selected = List<bool>.generate(listLength, (_) => false);
+  }
+
+  @override
+  void dispose() {
+    _selected.clear();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        appBar: AppBar(
+          title: const Text(
+            'ListTile select example',
+          ),
+          leading: isEditMode
+              ? IconButton(
+                  icon: const Icon(Icons.close),
+                  onPressed: () {
+                    setState(() {
+                      isEditMode = false;
+                    });
+                  },
+                )
+              : const BackButton(),
+          actions: <Widget>[
+            if (_isGridMode)
+              IconButton(
+                icon: const Icon(Icons.grid_on),
+                onPressed: () {
+                  setState(() {
+                    _isGridMode = false;
+                  });
+                },
+              )
+            else
+              IconButton(
+                icon: const Icon(Icons.list),
+                onPressed: () {
+                  setState(() {
+                    _isGridMode = true;
+                  });
+                },
+              ),
+            if (isEditMode)
+              TextButton(
+                  child: !_selectAll
+                      ? const Text('select all')
+                      : const Text('unselect all'),
+                  onPressed: () {
+                    _selectAll = !_selectAll;
+                    setState(() {
+                      _selected =
+                          List<bool>.generate(listLength, (_) => _selectAll);
+                    });
+                  }),
+          ],
+        ),
+        body: _isGridMode
+            ? GridBuilder(
+                isEditMode: isEditMode,
+                selectedList: _selected,
+                onEditChange: (bool x) {
+                  setState(() {
+                    isEditMode = x;
+                  });
+                },
+              )
+            : ListBuilder(
+                isEditMode: isEditMode,
+                selectedList: _selected,
+                onEditChange: (bool x) {
+                  setState(() {
+                    isEditMode = x;
+                  });
+                },
+              ));
+  }
+}
+
+// ignore: must_be_immutable
```

Comment:
Can we avoid using ignores in sample code? We consider ignores in code as technical debt, things that need to be fixed. Since this is meant to be instructive it probably should not have an ignore.

Reply:
updated code sample